### DISCUSSION TITLE
Something went wrong with multiple checkboxes

### DISCUSSION THREAD

**Discussion by andrconstruction** [react-hook-form/react-hook-form#10798] [General] [answered]
Please explain me what I did wrong, if you can.
I have a form like this

Code example.

[https://codesandbox.io/p/sandbox/react-hook-form-3pygzt](https://codesandbox.io/p/sandbox/react-hook-form-3pygzt)

Output data is expected and work perfectly But if I added defaultValues or use reset() . output values is inconsitence

**Comment by leapful** [comment]
@andrconstruction 

Please describe more detail on what is wrong after adding default values or reset method.

**Reply by andrconstruction** [reply]
if I added default values then within an output data disappear true values. In the code example (index.js  row 126)  you can remove comment on default values  and will see the broken output

**Comment by Moshyfawn** [comment, marked as answer]
You're using a **"flat array"** which [is **not supported**](https://react-hook-form.com/docs/usefieldarray#rules) by RHF.

![rhf-flat-array](https://github.com/react-hook-form/react-hook-form/assets/16290753/db678462-cfca-431f-9d36-0214268f314d)

You need to provide a field name for the array instead.

```diff
useForm({
- defaultValues: defaultSubscribes
+ defaultValues: {
+  subscribers: defaultSubscribes
+ }
})

<CheckboxNg
- name={`${index}.email`}
+ name={`subscribers.${index}.email`}
  control={control}
/>
```

Here's [a working codesandbox](https://codesandbox.io/p/sandbox/reverent-christian-9fvntl?file=%2Fsrc%2Findex.js%3A126%2C39).

**Reply by andrconstruction** [reply]
Thank you
